source/components/atoms/Select/Select.tsx

```diff
@@ -43,59 +43,74 @@ interface Props {
   style?: ViewStyle;
   onBlur: (value: string) => void;
   onOpen: (event: undefined, isSelect: true) => void;
+  onClose: (event: undefined, isSelect: true) => void;
   showErrorMessage?: boolean;
   error?: { isValid: boolean; message: string };
 }
 
-const Select: React.FC<Props> = React.forwardRef(({
-  items,
-  onValueChange,
-  onBlur,
-  onOpen,
-  placeholder,
-  value,
-  editable = true,
-  showErrorMessage = true,
-  error,
-  style,
-}, ref) => {
+const Select: React.FC<Props> = React.forwardRef(
+  (
+    {
+      items,
+      onValueChange,
+      onBlur,
+      onOpen,
+      onClose,
+      placeholder,
+      value,
+      editable = true,
+      showErrorMessage = true,
+      error,
+      style,
+    },
+    ref
+  ) => {
+    const currentItem = items.find((item) => item.value === value);
+    const handleValueChange = (itemValue: string | number | boolean) => {
+      if (onValueChange && typeof onValueChange === "function") {
+        onValueChange(itemValue ? itemValue.toString() : null);
+      }
 
-  const currentItem = items.find(item => item.value === value);
-  const handleValueChange = (itemValue: string | number | boolean) => {
-    if (onValueChange && typeof onValueChange === 'function') {
-      onValueChange(itemValue ? itemValue.toString() : null);
-    }
+      if (itemValue == currentItem?.value) {
```

Comment:
I guess we can remove one if-statement and write it like this instead:

```js
if (itemValue !== currentItem?.value) {
  onBlur(currentItem.value)
}
```

We do not need to check for onBlur since its a required property (according to props interface)